runtime_api.go

```diff
@@ -0,0 +1,225 @@
+package main
+
+import (
+	"go/ast"
+	"go/token"
+)
+
+// addRuntimeAPI exposes additional functions in the runtime
+// package that may be helpful when hiding information
+// during execution is required.
+func addRuntimeAPI(filename string, file *ast.File) {
+	switchPanicPrints := func(node ast.Node) bool {
+		call, ok := node.(*ast.CallExpr)
+		if !ok {
+			return true
+		}
+		id, ok := call.Fun.(*ast.Ident)
+		if !ok {
+			return true
+		}
+
+		if id.Name == "print" {
+			id.Name = "panicprint"
+			return false
+		}
+
+		return true
+	}
+
+	switch filename {
+	case "debug.go":
+		// Add HidePanics function and internal hidePanics variable
+		file.Decls = append(file.Decls, hidePanicsDecls...)
+	case "error.go":
+		// Add a function panicprint, that does nothing if panics are
+		// hidden, otherwise forwards arguments to printany to print them
+		// as normal.
```

Comment:
I don't get this. If we already modify printany to add an if statement to do nothing when "hide panics" is enabled, why do we need the extra panicprint func?

Reply:
So `panicprint` replaces `print` calls exclusively. `print` is a variadic function, so it must be replaced by a variadic funcion, hence the use of `panicprint`. The if statement is added to `printany` because it is called a handful of times directly when printing panics. 90% of the time `print` is called, but I need to handle the case when `printany` is as well.